### PR TITLE
Enabled handling of empty RG, LAW and AA

### PR DESCRIPTION
**Pull request by Rewdie**
Fix: Enabled handling of empty RG, LAW and AA by adding If-clauses

### CODE REVIEW

File: Scripts/main.ps1 [hunk 1]
```diff
@@ -11,20 +11,32 @@ Set-GitHubEnv -Name "EnvironmentID" -Value $EnvironmentID -Verbose
 Write-Output "$Task - Looking for monitoring Resource Group (RG):"
 $RGs = az group list -o json | ConvertFrom-JSON
 $RG = $RGs | Where-Object Name -match "$env:EnvironmentID-log"
-Set-GitHubEnv -Name "Monitoring_RGName" -Value $RG[0].name -Verbose
-Set-GitHubEnv -Name "Monitoring_RGID" -Value $RG[0].id -Verbose
+if($RG -ne $null){
```
Comment (MariusStorhaug) [inline Scripts/main.ps1]: Put $null on the left side of the comparrison

File: Scripts/main.ps1 [hunk 2]
```diff
@@ -11,20 +11,32 @@ Set-GitHubEnv -Name "EnvironmentID" -Value $EnvironmentID -Verbose
...
+}else{
+    Write-Output "$Task - No Resource Group (RG) found named $env:EnvironmentID-log in subscription $env:SubscriptionName"
+}
 
 Write-Output "$Task - Looking for monitoring Log Analytics workspace (LAW):"
 $LAWs = az monitor log-analytics workspace list -o json | ConvertFrom-JSON
 $LAW = $LAWs | Where-Object Name -match "$env:EnvironmentID-log"
-Set-GitHubEnv -Name "Monitoring_LAWName" -Value $LAW[0].name -Verbose
-Set-GitHubEnv -Name "Monitoring_LAWID" -Value $LAW[0].id -Verbose
+if($LAW -ne $null){
```
Comment (MariusStorhaug) [inline Scripts/main.ps1]: Put $null on the left side of the comparrison

File: Scripts/main.ps1 [hunk 3]
```diff
@@ -11,20 +11,32 @@ Set-GitHubEnv -Name "EnvironmentID" -Value $EnvironmentID -Verbose
...
+    Write-Output "$Task - No Log Analytics Workspace (LAW) found in subscription $env:SubscriptionName"
+}
 
 Write-Output "$Task - Looking for monitoring Automation Account(AA):"
 az config set extension.use_dynamic_install=yes_without_prompt
 $AAs = az automation account list -o json | ConvertFrom-JSON
 $AA = $AAs | Where-Object Name -match "$env:EnvironmentID-log-AA"
-Set-GitHubEnv -Name "Monitoring_AAName" -Value $AA[0].name -Verbose
-Set-GitHubEnv -Name "Monitoring_AAID" -Value $AA[0].id -Verbose
+if($AA -ne $null){
```
Comment (MariusStorhaug) [inline Scripts/main.ps1]: Put $null on the left side of the comparrison

**Review by MariusStorhaug — DISMISSED** [review]
I think the rest of the code is ok with some minor adjustments to the checks in the if statements

File: Scripts/main.ps1 (hunk 2 above)
Comment (Rewdie) [reply Scripts/main.ps1]: Fixed by removing comparison, now only checking if($LAW)

File: Scripts/main.ps1 (hunk 3 above)
Comment (Rewdie) [reply Scripts/main.ps1]: Fixed by removing comparison, now only checking if($AA)

File: Scripts/main.ps1 (hunk 1 above)
Comment (Rewdie) [reply Scripts/main.ps1]: Fixed by removing comparison, now only checking if($RG)

**Review by MariusStorhaug — APPROVED** [review]
This is also a good solution to the issue. Good job!

**Pull request merged**